src/main/java/teammates/ui/controller/InstructorFeedbackEditPageData.java

```diff
@@ -12,68 +13,400 @@
+        List<FeedbackQuestionTableRow> copyQuestionRows = buildCopyQuestionsModalRows(copiableQuestions,
+                                                                                  instructor);
+        copyQnForm = new FeedbackQuestionCopyTable(feedbackSession.courseId, feedbackSession.feedbackSessionName, 
+                                                   copyQuestionRows);
+    }
+
+
+    private List<FeedbackQuestionTableRow> buildCopyQuestionsModalRows(List<FeedbackQuestionAttributes> copiableQuestions,
+                                                                       InstructorAttributes instructor) {
+        List<FeedbackQuestionTableRow> copyQuestionRows = new ArrayList<FeedbackQuestionTableRow>();
+        if (instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION)) {
+            for (FeedbackQuestionAttributes question1 : copiableQuestions) {
+                String courseId = question1.courseId;
+                String fsName = question1.feedbackSessionName;
+                String qnType = question1.getQuestionDetails().getQuestionTypeDisplayName();
+                String qnText = question1.getQuestionDetails().questionText;
+                String qnId = question1.getId();
+                
+                FeedbackQuestionTableRow row = new FeedbackQuestionTableRow(courseId, fsName, qnType, qnText, qnId);
+                copyQuestionRows.add(row);
+            }
+        }
+        return copyQuestionRows;
+    }
+
+
+    private void buildPreviewForm(FeedbackSessionAttributes feedbackSession,
+                                    List<StudentAttributes> studentList,
+                                    List<InstructorAttributes> instructorList) {
+        previewForm = new FeedbackSessionPreviewForm(feedbackSession.courseId, feedbackSession.feedbackSessionName, 
+                                                     getPreviewAsStudentOptions(studentList), 
+                                                     getPreviewAsInstructorOptions(instructorList));
+    }
+
+
+    private void buildExistingQuestionForm(FeedbackSessionAttributes feedbackSession,
+                                           List<FeedbackQuestionAttributes> questions,
+                                           Map<String, Boolean> questionHasResponses,
+                                           InstructorAttributes instructor, FeedbackQuestionAttributes question) {
+        FeedbackQuestionEditForm qnForm = new FeedbackQuestionEditForm();
+        qnForm.setAction(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACK_QUESTION_EDIT));
+        qnForm.setCourseId(instructor.courseId);
+        
+        qnForm.setFeedbackSessionName(feedbackSession.feedbackSessionName);
+        qnForm.setQuestion(question);
+        qnForm.setQuestionDetails(question.getQuestionDetails());
+        qnForm.setQuestionNumberSuffix("-" + String.valueOf(question.questionNumber));
```

Comment:
The `int` will automatically be casted into a `String` here even without `String.valueOf()`.